resources/secrets/src/lib.rs

```diff
@@ -2,25 +2,36 @@ use std::collections::BTreeMap;
 
 use async_trait::async_trait;
 
-use shuttle_service::{Error, Factory, ResourceBuilder};
+use serde::{Deserialize, Serialize};
+use shuttle_service::{Error, Factory, ResourceBuilder, Type};
 
+#[derive(Serialize)]
 pub struct Secrets;
 
 /// Get a store with all the secrets available to a deployment
 #[async_trait]
 impl ResourceBuilder<SecretStore> for Secrets {
+    const TYPE: Type = Type::Secrets;
+
+    type Output = SecretStore;
+
     fn new() -> Self {
         Self {}
     }
 
-    async fn build(self, factory: &mut dyn Factory) -> Result<SecretStore, Error> {
+    async fn build(build_data: &Self::Output) -> Result<SecretStore, crate::Error> {
+        Ok(build_data.clone())
+    }
+
+    async fn output(self, factory: &mut dyn Factory) -> Result<Self::Output, crate::Error> {
         let secrets = factory.get_secrets().await?;
 
         Ok(SecretStore { secrets })
     }
 }
 
 /// Store that holds all the secrets available to a deployment
+#[derive(Deserialize, Serialize, Clone)]
```

Comment:
Since this is also the `Output` we'll end up storing the secret values in the DB. I don't know if we want to do that though :man_shrugging: